src/netaddress.h

```diff
@@ -192,6 +192,17 @@ class CNetAddr
         friend class CSubNet;
 
     private:
+        /**
+         * Check whether a container begins with the given prefix.
+         */
+        template <typename T1, size_t PREFIX_LEN>
```

Comment:
Yes, I did consider making it more visible but refrained because of the wretched `std::array` argument (which was imposed by gcc 4.8) - I think it is not very friendly like this.

Moved now to `util/string.h` since you mention this. Once support for gcc 4.8 is dropped we can improve it to take some more generic argument.